### PR TITLE
cleaning of sidebar UI

### PR DESCRIPTION
**Pull request by nsklaus**
### How to use
there's nothing to do, it's just a reordering and alignment of sidebar info elements. 

#### Summary
```SUMMARY: Interface "Cleaning of sidebar UI"```

#### Purpose of change
sidebar UI elements were not properly aligned and ordered.
please see attached "before / after" animated screenshots.

#### Describe the solution
reordering and alignment of sidebar info elements. 

#### Additional context
bleh

### CODE REVIEW

File: src/sidebar.cpp [hunk 1]
```diff
@@ -305,28 +305,39 @@ void player::disp_status( const catacurses::window &w, const catacurses::window
...
+        mvwprintz( weapwin, 0, x, style_color, style );
     }
 
-    wmove( w, sideStyle ? 1 : 2, 0 );
+    std::string hunger_string = "";
+    nc_color hunger_color = c_yellow;
     if( get_hunger() >= 300 && get_starvation() > 2500 ) {
-        wprintz( w, c_red,    _( "Starving!" ) );
+        hunger_color = c_red;
+        hunger_string = "Starving!";
```
Comment (AMurkin) [inline src/sidebar.cpp]: Why remove _(translation) here and at the next lines?

**Comment by Pupsi-Mupsi** [comment]
Love it. Translation will be a pain in the ass though - keeping the alignment and meaning together and so on...

Side by side view - made it easier for me to see the differences:
grafik
grafik

File: src/game.cpp [hunk 2]
```diff
@@ -237,7 +237,7 @@ game::game() :
     events( *event_manager_ptr ),
     weather( WEATHER_CLEAR ),
     lightning_active( false ),
-    u_shared_ptr( &u, null_deleter{} ),
+    u_shared_ptr( &u, null_deleter {} ),
```
Comment (jbytheway) [inline src/game.cpp]: You've made lots of unrelated whitespace changes like this one.  Please revert those.  They make the PR harder to review, and don't match the project style.

File: src/game.cpp [hunk 3]
```diff
@@ -875,22 +879,22 @@ bool game::start_game()
...
     const bool spawn_near =
-        get_option<bool>( "BLACK_ROAD" ) || g->scen->has_flag( "SUR_START" );
+        get_option<bool> ( "BLACK_ROAD" ) || g->scen->has_flag( "SUR_START" );
     // Surrounded start ones
     if( spawn_near ) {
         start_loc.surround_with_monsters( omtstart, mongroup_id( "GROUP_ZOMBIE" ), 70 );
     }
 
-    m.spawn_monsters( !spawn_near ); // Static monsters
+    m.spawn_monsters( !spawn_near );  // Static monsters
```
Comment (jbytheway) [inline src/game.cpp]: Other unrelated whitespace changes are like this one.

File: src/game.cpp (hunk 2 above)
Comment (nsklaus) [reply src/game.cpp]: it's not me, i haven't touched this line personally, it's the 'make astyle'.

File: src/game.cpp (hunk 3 above)
Comment (nsklaus) [reply src/game.cpp]: same thing, i have not altered this specific line myself. it's the automated 'make astyle' command.
i'm afraid i cannot comply with your requests since i must use 'make astyle' to pass the gorgon-ghprb checks..

File: src/sidebar.cpp (hunk 1 above)
Comment (nsklaus) [reply src/sidebar.cpp]: sorry aboutthat, i had no idea this was related to translation..

**Comment by nsklaus** [comment]
> Love it. Translation will be a pain in the ass though - keeping the alignment and meaning together and so on...
thanks for the kind words :)
this is just a first pass, just trying to fix the existing without any major changes.
imho, the sidebar need overhaul, in code cleaning, and in the way it is presenting the info. 
i'm thinking about making toggle-able floating panels, the user could activate some, and disable others, as he prefer. 
as for the translation, yes, each translators will have to carefully choose words specifically if he want to keep the same kind of alignment in his own language. this is left to the talent of each translators ..

**Comment by Pupsi-Mupsi** [comment]
Well said. I agree and I'm sure I'll be able to come up with a good solution.

**Comment by jbytheway** [comment]
Thanks for fixing some of the unneeded spaces, but there are still quite a few more.  You can seem them easily in the "Files changed" tab above.

**Comment by nsklaus** [comment]
> Thanks for fixing some of the unneeded spaces, but there are still quite a few more. You can seem them easily in the "Files changed" tab above.
i will try to have a look but, please tell me, if this is not respecting this project style, how come i'm passing the gorgon-ghprb tests ? unless i'm mistaken this server is responsible to enforce the style consistency right ? also i'm telling you at least the two changes you pointed me to were not done by me. i suspect they were done automatically by the make astyle pass.
some maing me change it back manually here and now will not solve the issue. if 'make astyle' is producing this problem then it's 'make astyle' that need to get fixed.

**Comment by nsklaus** [comment]
> Thanks for fixing some of the unneeded spaces, but there are still quite a few more. You can seem them easily in the "Files changed" tab above.

yes i've been looking at "Files changed" tab and again i can confirm there's lots of those "space changes" you're talking about they weren't done by me. i suggest you (i mean this project) fix your problems with your scripts and not add this burden of manual fixing to those who want to help and contribute ? the only other reason i can think of is: when i tested a few cpp IDE, one of them made this. but i certainly didn't do it myself. this is the work of some automated pass.

**Comment by ZhilkinSerg** [comment]
You better call it:

```SUMMARY: Interface "Cleaning of sidebar UI"```

**Comment by cainiaowu** [comment]
Have you test with a very long name range weapon with charge?

**Comment by nsklaus** [comment]
> Have you test with a very long name range weapon with charge?

no, i fact i coded this blindfolded while i was asleep. in my dreams i could see some horizontal space being filled with letters. and it was displayed good in a sidebar. when i woke up there was this screenshot where one could see all the space available for the letters to go on that 'weapon' line .. amazing huh ?

**Comment by Pupsi-Mupsi** [comment]
> 
> 
> Have you test with a very long name range weapon with charge?

Could you give an example? Maybe @nsklaus will wield this weapon and post a screenshot for you.

**Comment by Pupsi-Mupsi** [comment]
> 
> 
> > Have you test with a very long name range weapon with charge?
> 
> no, i fact i coded this blindfolded while i was asleep. in my dreams i could see some horizontal space being filled with letters. and it was displayed good in a sidebar. when i woke up there was this screenshot where one could see all the space available for the letters to go on that 'weapon' line .. amazing huh ?

Indeed ;-) I do code "blind" at times.

**Comment by Pupsi-Mupsi** [comment]
I'm sorry. Hope I didn't offend you. I'm sure you know what you're doing. And I'm sure cainiaowu didn't mean no disrespect either. Keep up the good work.

BTW - since you are far more capable than I am. Could take a look at my latest PR. #27355
I sure to use some help with that.

**Comment by Night-Pryanik** [comment]
Please revert season/temp and weapon line changes. Weapon line should be the last.
Also, I don't like the basic idea of alignment based on only one language. For example, `lighting` in English is 8 letters, while `освещение` in Russian is 9 letters, and I don't want to truncate localized variant.

File: src/game.cpp [hunk 4]
```diff
@@ -2977,7 +2981,7 @@ void game::debug()
         _( "Teleport - Adjacent overmap" ),     // 31
         _( "Test trait group" ),                // 32
         _( "Show debug message" ),              // 33
-        _( "Crash game (test crash handling)" ),// 34
+        _( "Crash game (test crash handling)" ),  // 34
```
Comment (Night-Pryanik) [inline src/game.cpp]: This isn't needed.

File: src/game.cpp [hunk 5]
```diff
@@ -3619,10 +3625,13 @@ void game::draw_sidebar()
     if( sideStyle ) {
         werase( w_status2 );
     }
-    u.disp_status( w_status, w_status2 );
 
+    // sidestyle ? narrow_1 : wider_0
```
Comment (Night-Pryanik) [inline src/game.cpp]: Please remove the commented section.

File: src/game.cpp [hunk 6]
```diff
@@ -4038,7 +4062,7 @@ void game::draw_minimap()
 
     // Print arrow to mission if we have one!
     if( !drew_mission ) {
-        double slope = ( cursx != targ.x ) ? double( targ.y - cursy ) / double( targ.x - cursx ) : 4;
+        double slope = ( cursx != targ.x ) ? double( targ.y - cursy ) / double ( targ.x - cursx ) : 4;
```
Comment (Night-Pryanik) [inline src/game.cpp]: This isn't needed.

**Comment by nsklaus** [comment]
> Please revert season/temp and weapon line changes. Weapon line should be the last.
> Also, I don't like the basic idea of alignment based on only one language. For example, `lighting` in English is 8 letters, while `освещение` in Russian is 9 letters, and I don't want to truncate localized variant.

as i said, each translators will have to come up with words and arrangements that will display nicely in his own language. yes, that include russian.
also, i'd like to know why weapon line sould be the last ? 
please provide a good reason for me to comply with your request.

File: src/game.cpp (hunk 5 above)
Comment (nsklaus) [reply src/game.cpp]: why ? am i not allowed to leave comment  related to the code ?

**Comment by nsklaus** [comment]
> I mentioned that particular test case is because we have exactly that bug before.
> #25598

'normal' goes away if you equip a gun.

File: src/game.cpp (hunk 5 above)
Comment (Night-Pryanik) [reply src/game.cpp]: Oh, it is related to the code? Then leave it, I just thought it is a left-over.

**Comment by Night-Pryanik** [comment]
> also, i'd like to know why weapon line sould be the last ?
please provide a good reason for me to comply with your request.

The lines "Weather", "Lighting" and "Season" are related, united by the "Environment" theme (at least to me), so I think it's natural not to tear them apart with the "Weapon" line, which corresponds to the "Player" theme.

**Comment by nsklaus** [comment]
> > also, i'd like to know why weapon line sould be the last ?
> > please provide a good reason for me to comply with your request.
> 
> The lines "Weather", "Lighting" and "Season" are related, united by the "Environment" theme (at least to me), so I think it's natural not to tear them apart with the "Weapon" line, which corresponds to the "Player" theme.

i'm sorry but this is not a good reason enough for me to justify altering significantly the goal of this contribution.
the goal was UI clean up and alignment of elements. what you propose would break that.

i will not change it for this specific point you're asking.
so i see two way out of this:
- refuse the contribution, or
- accept it as it is and once committed you'll do what you want.

**Comment by Night-Pryanik** [comment]
> the goal was UI clean up and alignment of elements. what you propose would break that.

How could my variant

default

instead of your variant

default

break "UI clean up and alignment of elements"?

**Comment by nsklaus** [comment]
> > the goal was UI clean up and alignment of elements. what you propose would break that.
> 
> How could my variant
> 
> default
> 
> instead of your variant
> 
> default
> 
> break "UI clean up and alignment of elements"?

if this PR, which is small change, is a problem, what will happen if i spend time to implement the mockup i linked above ?

**Comment by Night-Pryanik** [comment]
> if this PR, which is small change, is a problem

It isn't a problem at all. 
The moments I'm making accents on is my personal opinion.

**Comment by nsklaus** [comment]
> @nsklaus That mockup would have to be a separate PR, which could possibly take a long time to review (the existence of translations makes UI changes take a loong time)

yes of course it would be another PR dedicated to the new sidebar as described in the mockup.
why i'm talking about that here? because i'd like to ensure i won't loose my time doing it.
if small changes happening in This particular PR are problematic, then the big changes i planned next will be even more problematic.

**Comment by nsklaus** [comment]
> > if this PR, which is small change, is a problem
> 
> It isn't a problem at all.
> The moments I'm making accents on is my personal opinion.

@Night-Pryanik thanks for telling. it's good to hear it's not a problem.
so if this PR gets accepted, you could always make new PR, to change things in the order you prefer and see if it gets accepted, and if so then it's all fine ? 
i've spent time to make it look like that. like what i thought would be better. 
and now i would like to focus on trying to make the new "sidebar".

**Comment by nsklaus** [comment]
> Do as you see fit. I'm not the one who will make decision on whether this PR is worth merging, I'm only one of the reviewers.

thank you for your honesty, and the time you take for helping with reviews.

**Comment by FrozenSnowFox** [comment]
While I agree with Night-Pryanik about the weapon being last I don't agree with shifting the date above, it looks weird.  What about this setup?  Side by side obviously, maybe with Moon on the same line as Spring.  At the end of the day you can't please everyone so do as you see fit, it's probably a moot point anyway if your proposed UI overhaul goes through.

Spring, day 2 [  +    ]
Location: office
Weather: rain
Lighting: bright
Weapon: crowbar

Moon: (     )
Sound: 0
Temp: 3C
Normal

**Comment by Crafty908** [comment]
> 
> 
> > the goal was UI clean up and alignment of elements. what you propose would break that.
> 
> How could my variant
> 
> default
> 
> instead of your variant
> 
> default
> 
> break "UI clean up and alignment of elements"?

I agree that weapon on the bottom makes the most sense as it is the applicator of change to the Zombie line below. The mind of the player should not have to engage in thinking "i have to look for my weapon, oh. Its above the Day" section. If the player has to look for information then the user interface is not better, it should be obvious at first glance where things are.
